### DISCUSSION TITLE
full sized window with async arg

### DISCUSSION THREAD

**Discussion by benschlegel** [johnlindquist/kit#1474] [Q&A] [answered]
I'm trying to use `arg()` with a dynamic list like described in the docs (`await arg("Select a Star Wars Character", async () => {...}`). The API I'm using takes around 600ms to respond, which triggers a pretty fast re-render of the arg window that's a bit annoying. 

Is there a way I can render the `arg`  window the maximum size even before the choices are loaded?

**Comment by johnlindquist** [comment]
@benschlegel have it initially respond with an array of placeholders/loading values

**Reply by benschlegel** [reply]
How can i do this? I tried doing something like `await arg({initialChoices: ["test", "test"]}, async () => {...}`, however, the `initialChoices` never seem to get overridden after the async function returns (which is the behavior I would have expected)

**Reply by JosXa** [reply]
I'm not sure this helps with your problem, but what John probably means is this:

```ts
await arg("Select a Star Wars Character", async () => {
  await setChoices([{ name: "Loading..." }])

  await wait(1500)

  return [{ name: "Yoda" }, { name: "Anakin" }]
})

```

**Reply by JosXa** [reply]
I still see a bit of flicker doing it with a setter though. Let me see...

**Reply by JosXa** [reply, marked as answer]
This is nice:
```ts
await arg(
  {
    placeholder: "Select a Star Wars Character",
    async onInit() {
      // Load data here
      await wait(1200)

      await setChoices([{ name: "Yoda" }, { name: "Anakin" }])
    },
  },
  [{ name: "Loading...", onSubmit: () => preventSubmit }],
)
```

**Reply by benschlegel** [reply]
Working exactly as I intended it to, thanks!!

**Reply by JosXa** [reply]
I believe what you ran into with the `arg({initialChoices: [...]}, async () => ...)` combination is that the async getter function actually only triggers on user input. It's supposed to be a `async (input?: string) => ...`:

```ts
await arg(
  {
    placeholder: "Select a Star Wars Character",
    initialChoices: [{ name: "Loading...", onSubmit: () => preventSubmit }],
  },
  (input) => {
    console.log("This will only be triggered when you type something!")
    return []
  },
)
```


I believe that is expected behavior, even if it violates the principle of least surprise, since we're already passing **initial** choices.
